apps/a_dndtoggle/settings.js

```diff
@@ -0,0 +1,31 @@
+(function(back) {
+
+    const settings = 
+        require('Storage').readJSON("a_dndtoggle.settings.json", true) || {};
+  
+    function updateSettings() {
+      require('Storage').writeJSON("a_dndtoggle.settings.json", settings);
+    }
+
+    function buildMainMenu(){
+      const modes = [/*LANG*/"Noisy",/*LANG*/"Alarms",/*LANG*/"Silent"];
+      let mainmenu = {
+        '': { 'title': 'A_DND Toggle' },
+        '< Back': back,
+        /*LANG*/"Quiet Mode": {
+          value: settings.mode || 2,
+          min: 1, max: modes.length - 1,
```

Comment:
To check I understand here, do we want mode to go between 1 (`Noisy`) and 3 (`Silent`)? Or do we want it to be 0-indexed, so 0 (`Noisy`) up to 3 (`Silent`) ?

Reply:
0 Noisy is just there for being a placeholder.
1 - Alarms, 2 - Silent are the actual choices, so min = 1 and max = 2.

Mode could be an object with {1:"Alarms",2:"Silent"} but the array pattern is what I saw in other spots, so I went with that.

Reply:
_(No rush, I'm just adding to make sure I explain the best I can. I probably should've documented better.)_

The Bangle quiet mode options are: `0-Off`, `1-Alarms`, `2-Silent`

The current a_dndtoggle app is hard coded to toggle between `0-Off` and `2-Silent`. 
- If current mode is `1-Alarms` or `2-Silent`, it goes to `0-Off`. 
- If current mode is `0-Off`, it goes to `2-Silent`.

This new settings page lets you pick if it should go to `1-Alarms` or `2-Silent`.

I think my options were:
1. Have an array of `["Alarms", "Silent"]`, then add `1` to the result
2. Have an array of `[null,"Alarms","Silent"]`, and not allow picking `0`

I went with option 2, but instead of "null" I left it labelled as Noisy.

I've added a comment in settings.js and a note in the readme about the settings page options.

Let me know if I should add further comments, or if there's a better way.

Reply:
No that's not a problem, it all makes sense! Thanks for adding the comments though, all very clear now - option 2 is easier for us to map across to the bangle options, or at least in my head :)